src/Microsoft.AspNet.OData.Shared/Builder/StructuralTypeConfiguration.cs

```diff
@@ -58,7 +57,9 @@ protected StructuralTypeConfiguration(ODataModelBuilder modelBuilder, Type clrTy
             ClrType = clrType;
             ModelBuilder = modelBuilder;
             _name = clrType.EdmName();
-            _namespace = clrType.Namespace ?? DefaultNamespace;
+
+            // Use the namespace is uder specified one, otherwise use CLR Namespace. If CLR Namespace is null we fallback to "Default"
```

Comment:
What's the scenario about "CLR namespace is null?"

Reply:
No realistic scenario tbh. I left it there thinking there might have been scenario. According to documentation it can be null for anonymous types and generic params both of which I don't think are allowed by OData. If you think this is safe to remove I can remove. 

https://docs.microsoft.com/en-us/dotnet/api/system.type.namespace?redirectedfrom=MSDN&view=netframework-4.7.2#System_Type_Namespace